pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -341,7 +343,7 @@ def is_optimized_sync(repository, remote, url):
     return is_optimized
 
 
-def synchronize(remote_pk, repository_pk, mirror, skip_types, optimize):
+def synchronize(remote_pk, repository_pk, mirror, skip_types, optimize, url=None):
```

Comment:
should not you pass url into the RpmFirstStage?

Reply:
no need as there is already some logic to check and pass URL to first stage:
https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/tasks/synchronizing.py#L421 & [#L447](https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/tasks/synchronizing.py#L447) & [#L465](https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/tasks/synchronizing.py#L465) & [#L482](https://github.com/pulp/pulp_rpm/blob/master/pulp_rpm/app/tasks/synchronizing.py#L482)
so it is enough to process it right in sync itself.